src/components/input-text/input-text.tsx

```diff
@@ -0,0 +1,554 @@
+    this.previousEmittedValue = newPreviousEmittedValue;
+  };
+
+  private setPreviousValue = (newPreviousValue: string): void => {
+    this.previousValue = newPreviousValue;
+  };
+
+  private setValue = ({
+    committing = false,
+    nativeEvent,
+    origin,
+    previousValue,
+    value
+  }: {
+    committing?: boolean;
+    nativeEvent?: MouseEvent | KeyboardEvent | InputEvent;
+    origin: SetValueOrigin;
+    previousValue?: string;
+    value: string;
+  }): void => {
+    this.setPreviousValue(previousValue || this.value);
+    this.previousValueOrigin = origin;
+    this.userChangedValue = origin === "user";
+    this.value = value;
+
+    if (origin === "direct") {
+      this.setInputValue(value);
+    }
+
+    if (nativeEvent) {
+      const calciteInputTextInputEvent = this.calciteInputTextInput.emit({
+        element: this.childEl,
+        nativeEvent,
+        value: this.value
+      });
+
+      if (calciteInputTextInputEvent.defaultPrevented) {
+        this.value = this.previousValue;
+      } else if (committing) {
+        this.emitChangeIfUserModified();
+      }
+    }
+  };
+
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  render(): VNode {
+    const dir = getElementDir(this.el);
+    const loader = (
+      <div class={CSS.loader}>
+        <calcite-progress label={this.intlLoading} type="indeterminate" />
+      </div>
+    );
+
+    const inputClearButton = (
+      <button
+        aria-label={this.intlClear || TEXT.clear}
+        class={CSS.clearButton}
+        disabled={this.disabled || this.readOnly}
+        onClick={this.clearInputValue}
+        tabIndex={this.disabled ? -1 : 0}
+        type="button"
+      >
+        <calcite-icon icon="x" scale="s" />
+      </button>
+    );
+    const iconEl = (
+      <calcite-icon
+        class={CSS.inputIcon}
+        flipRtl={this.iconFlipRtl}
+        icon={this.requestedIcon}
+        scale="s"
+      />
+    );
+    const prefixText = <div class={CSS.prefix}>{this.prefixText}</div>;
+
+    const suffixText = <div class={CSS.suffix}>{this.suffixText}</div>;
+
+    const childEl = (
+      <input
+        aria-label={getLabelText(this)}
+        autofocus={this.autofocus ? true : null}
+        class={{
+          [CSS.editingEnabled]: this.editingEnabled,
+          [CSS.inlineChild]: !!this.inlineEditableEl
+        }}
+        defaultValue={this.defaultValue}
+        disabled={this.disabled ? true : null}
+        enterKeyHint={this.el.enterKeyHint}
+        inputMode={this.el.inputMode}
+        maxLength={this.maxLength}
+        minLength={this.minLength}
+        name={this.name}
+        onBlur={this.inputBlurHandler}
+        onFocus={this.inputFocusHandler}
+        onInput={this.inputInputHandler}
+        onKeyDown={this.inputKeyDownHandler}
```

Comment:
I wonder since this are all internal, if they should be named `internalInput*Handler`?

Reply:
I can't find any other examples of using the `internal` prefix for handlers in our repo, are you suggesting a convention change? I think its unnecessarily verbose

Reply:
Not a huge deal, but perhaps it could be a good convention change.  It would differentiate between internal handlers when doing project-wide searches.  Not a blocker for this pr though.

Reply:
Are there any external handlers?

Reply:
Not necessarily in tsx, but there would be in usage snippets like in tests and docs/html.